cunumeric/module.py

```diff
@@ -3047,7 +3170,7 @@ def _contract(
     for (mode, count) in c_b_modes.items():
         if count > 1:
             axes = [i for (i, m) in enumerate(b_modes) if m == mode]
-            b = b._diag_helper(axes=axes)
+            b = b._diag_helper(axes=axes)  # type: ignore [union-attr]
```

Comment:
not sure what's going on here yet. Another case I'd rather come back to in a dedicated pass that looks at all the existing ignores outside all this larger work.